fastify.js

```diff
@@ -88,8 +88,13 @@ function fastify (options) {
 
   validateBodyLimitOption(options.bodyLimit)
 
+  const caseSensitiveParse = (string) => {
+    const shouldLower = options.caseSensitive === false
+    return querystring.parse(shouldLower ? string.toLowerCase() : string)
```

Comment:
I'm a bit worried, is it expected that value will be lowered as well as key?

```
> require('querystring').parse('name=Anr&nAmE=LL&namE=Noda'.toLowerCase())
[Object: null prototype] { name: [ 'anr', 'll', 'noda' ] }
```

Reply:
Good catch, we don't want this

Reply:
So what is option here then?

Reply:
solution would be something like: 
```js
const caseSensitiveParse = (string) => {
  const shouldLower = options.caseSensitive === false
  let out = querystring.parse(string)
  if(shouldLower) {
    out = Object.keys(out).reduce((acc, key) => { acc[key.toLowerCase()] = out[key]; return acc }, {})
  }
  return out
}
```

Reply:
Yeah, but it does not take in account cases when query is `name=1&name=2` for example

Reply:
So seems like we have to make our own solution and cover all the places to add option for case-insensetive query parsing

Reply:
What about this?

```
const caseSensitiveParse = (string) => {
    const shouldLower = options.caseSensitive === false
    let out = querystring.parse(string)

    if (shouldLower) {
      out = Object.keys(out).reduce((acc, key) => {
        const lowerKey = key.toLowerCase()
        const existingItem = acc[lowerKey]

        // If a variation of the key exists, merge both
        if (existingItem) {
          acc[lowerKey] = [existingItem, out[key]].flat()
        } else {
          acc[lowerKey] = out[key]
        }

        return acc
      },
      {})
    }
    return out
  }
```

Reply:
> Yeah, but it does not take in account cases when query is `name=1&name=2` for example

so in the case of a querystring like `name=Anr&nAmE=LL&namE=Noda` the solution of @Nikodermus would work and output `name: ['Anr','LL','Noda']`

Reply:
I'm almost ok with that solution but we should take in account that fastify supports Node 10+ and Node 10 does not support [`Array.prototype.flat`](https://node.green/#ES2019-features-Array-prototype--flat--flatMap-)

Reply:
Made an update, using reduce instead of .flat